src/Cache.php

```diff
@@ -78,6 +78,8 @@ public function removeLegacyTags(array $data): array
             foreach ($versions as $version => $composerJson) {
                 if ('dev-master' === $version) {
                     $normalizedVersion = $this->versionParser->normalize($devMasterAlias);
+                } elseif (!isset($composerJson['version_normalized'])) {
```

Comment:
why not attempting to read `$composerJson['version']`  and normalizing it (or just `$version`) ? We have access to the VersionParser to normalize here.

Reply:
because we don't really care: when this happens, we always want to keep the "version" anyway